runtime/common/src/lib.rs

```diff
@@ -21,24 +21,51 @@
 mod apis;
 
 pub use frame_support::{
+    dispatch::DispatchClass,
     parameter_types,
     traits::{Currency, OnUnbalanced},
-    weights::constants::WEIGHT_PER_SECOND,
+    weights::{
+        constants::{BlockExecutionWeight, ExtrinsicBaseWeight, WEIGHT_PER_SECOND},
+        Weight,
+    },
 };
+use frame_system::limits::{BlockLength as SysBlockLength, BlockWeights as SysBlockWeights};
 use runtime_primitives::{AccountId, Balance, BlockNumber};
 use sp_runtime::{Perbill, Percent};
 
-// Extrinsics with DispatchClass::Normal only account for user messages
-// TODO: consider making the normal extrinsics share adjustable in runtime
+/// We assume that ~10% of the block weight is consumed by `on_initialize` handlers.
+/// This is used to limit the maximal weight of a single extrinsic.
+const AVERAGE_ON_INITIALIZE_RATIO: Perbill = Perbill::from_percent(10);
+/// We allow `Normal` extrinsics to fill up the block up to 25%.
+/// Such extrinsics only account for user messages. The rest of the block time can be used
+/// for message queue processing and/or `Operatoinal` extrinsics.
+// TODO: consider making the normal extrinsics share adjustable
 const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(25);
+/// We allocate one third of the average block time for compute.
+const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND.saturating_div(3);
```

Comment:
This won't be correct for vara runtime btw, so I'd suggest to remove this from common